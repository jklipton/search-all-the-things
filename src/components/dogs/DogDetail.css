@import url('https://fonts.googleapis.com/css?family=Open+Sans:700,800i');

:local(.detail) {
    text-align: center;
    overflow: auto;
      
    img {
        height: 30vw;
        border: 5px solid black;
        margin: 10px;
        box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
      }

    p {
        margin: .2em;
    }

    p:last-child{
        margin-top: 1em;
    }

}