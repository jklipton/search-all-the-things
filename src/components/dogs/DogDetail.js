import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getDog } from '../../../services/dogApi';
import styles from './DogDetail.css';

export default class DogDetail extends Component {

  static propTypes = {
    dogID: PropTypes.string.isRequired,
    history: PropTypes.object
  };

  state = {
    dog: ''
  };

  componentDidMount() {
    getDog(this.props.dogID)
      .then(({ petfinder }) => this.setState({ dog: petfinder.pet }));
  }

  handleBack = event => {
    event.preventDefault();
    event.stopPropagation();
    this.props.history.goBack();
  };

  parsePhoto = (media) => {
    if(!media.photos) return;

    return media.photos.photo.find((photo) => {
      return photo['@size'] === 'x';
    });
  };

  render() {
    const { dog } = this.state;
    let photo = null;
    let breeds = null;
    if(dog){ 
      photo = dog.media.photos ? this.parsePhoto(dog.media).$t : null;
      const isArray = dog.breeds.breed[0];
      breeds = isArray ? (
        dog.breeds.breed.map(breed => `${breed.$t},`)
      ) : (
        dog.breeds.breed.$t
      );
    }

    if(dog === null) return null;

    return (
      <article>
        <div className={styles.detail}>
          {/* <a href="" onClick={this.handleBack}>Back</a> */}
          <img src={photo}/>
          <h2>{dog.name.$t}</h2>
          <p>Age : {dog.age.$t} | Size: {dog.size.$t} | Sex: {dog.sex.$t}</p>
          <p>
            Breeds : {breeds}
          </p>
          <p> Location: {dog.contact.state.$t}, {dog.contact.zip.$t} </p>
          <p>
            {dog.description.$t}
          </p>
        </div>
      </article>
    );
  }
}