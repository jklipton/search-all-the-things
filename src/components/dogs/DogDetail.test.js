import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DogDetail from './DogDetail';

describe('DogDetail', () => {

  it('renders as designed', () => {
    const wrapper = shallow(<DogDetail dogID={null} history={null}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders with id as designed', () => {
    const wrapper = shallow(<DogDetail dogID={'41724600'} history={null}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

});
