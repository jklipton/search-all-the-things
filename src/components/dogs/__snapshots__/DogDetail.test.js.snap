// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogDetail renders as designed 1`] = `""`;

exports[`DogDetail renders with id as designed 1`] = `""`;
